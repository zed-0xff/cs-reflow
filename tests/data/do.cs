void case1()
{
    init();
    do
    {
        if (y)
        {
            foo();
        }
        else
        {
            bar();
        }
    } while (x);
}

void case2()
{
    bool y = true;
    do
    {
        if (y)
        {
            foo();
        }
        else
        {
            bar();
        }
    } while (x);
}

void case3()
{
    bool x = false;
    do
    {
        if (y)
        {
            foo();
        }
        else
        {
            bar();
        }
    } while (x);
}

void case4()
{
    int i = 0;
    do
    {
        switch(i)
        {
            case 0:
                foo();
                break;
            case 1:
                bar();
                break;
            case 2:
                return;
        }

        i++;
    } while (true);
}

void case5()
{
    int i = 0;
    while (i<2)
    {
        do
        {
            if (i==0)
            {
                foo();
            }
            else
            {
                bar();
            }
        }
        while (false);

        i++;
    }
}
