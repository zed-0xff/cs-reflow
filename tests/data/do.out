void case1()
{
    init();  // InvocationExpression is not supported.
    if (y)
    {
        foo();
    }
    else
    {
        bar();
    }
}

void case2()
{
    foo();  // InvocationExpression is not supported.
}

void case3()
{
    if (y)
    {
        foo();
    }
    else
    {
        bar();
    }
}

void case4()
{
    foo();  // InvocationExpression is not supported.
    bar();  // InvocationExpression is not supported.
}

void case5()
{
    foo();  // InvocationExpression is not supported.
    bar();  // InvocationExpression is not supported.
}

